Lib/test/test_importlib/test_api.py

```diff
@@ -354,6 +355,20 @@ def test_module_missing_spec(self):
             with self.assertRaises(ModuleNotFoundError):
                 self.init.reload(module)
 
+    def test_reload_traceback_with_non_str(self):
+        # gh-125519
+        with support.captured_stdout() as stdout:
+            try:
+                self.init.reload("typing")
+            except TypeError as exc:
+                traceback.print_exception(exc, file=stdout)
+            else:
+                self.fail("Expected TypeError to be raised")
+        printed_traceback = stdout.getvalue()
+        self.assertIn("TypeError", printed_traceback)
+        self.assertNotIn("AttributeError", printed_traceback)
+        self.assertNotIn("module.__spec__.name", printed_traceback)
```

Comment:
Could also do this with assertions on the `__cause__` or `__context__` attribute of the raised exception, but that feels like an implementation detail to me. I prefer for the test to assert what the user will be seeing when the exception is raised (which feels like the important thing).